### PR TITLE
Revert "Merge pull request #2274 from steemit/revert-2235"

### LINKED ISSUE #2203: Test components
Figure out how to test components, and set up some examples.

### PR DESCRIPTION
**Pull request by bnchdrff**
closes #2203

This reverts commit 1c1451275e1f22f55af7cf6c7ae39e427c11e67a, reversing
changes made to a665b02e812469051a1aae5f4bbdcfb8434cb349.

### CODE REVIEW

**Comment by gl2748** [comment]
Looking at this now...

**Comment by gl2748** [comment]
For whatever reason, if you have a locally built version of the site at `/lib` the tests will run against that code and fail. When testing locally and checking our this branch for the first time, delete `lib` dir:
Example error:
```
 FAIL  lib/shared/HtmlReady.test.js
  ● Test suite failed to run

    Cannot find module 'chai' from 'HtmlReady.test.js'

       5 | var _from2 = _interopRequireDefault(_from);
       6 | 
    >  7 | var _chai = require('chai');
       8 | 
       9 | var _chai2 = _interopRequireDefault(_chai);
      10 | 
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:194:17)
      at Object.<anonymous> (lib/shared/HtmlReady.test.js:7:13)

```

**Comment by bnchdrff** [comment]
we need to exclude `lib/` from jest; i'll work on that

**Comment by bnchdrff** [comment]
oh
actually.... shouldn't the tests pass from `lib` anyways? is your `lib` build stale?

**Comment by bnchdrff** [comment]
the compiled tests pass for me.. try `yarn build && yarn test`

**Comment by gl2748** [comment]
Yarn run test should probably run build if it's testing the `/lib` dir?

**Comment by bnchdrff** [comment]
jest will run any *.test.js it finds -- if you've got a build in `lib` it'll run them in there also -- i think this is good

**Comment by gl2748** [comment]
I think we should add a jest config to this work. I think a reasonable expectation is that tests are running against one version of the code and that version is what's in `/src`.
https://facebook.github.io/jest/docs/en/configuration.html

**Comment by gl2748** [comment]
https://facebook.github.io/jest/docs/en/configuration.html#rootdir-string

**Comment by bnchdrff** [comment]
@gl2748 ok we will avoid testing `lib/` -- i changed our jest config

**Comment by gl2748** [comment]
@bnchdrff Actually, am i right in this, we had issues with Mocha not testing code as compiled ... perhaps testing lib is the right way to go after all?

**Comment by bnchdrff** [comment]
if we want to mandate testing of transpiled code in `lib` we can do that in another PR; the status quo is to let the test runner transpile when it runs

**Comment by gl2748** [comment]
Ok, sounds alright. Here is the issue that referenced the divergence between our code and the tests:
https://github.com/steemit/condenser/issues/2167

File: package.json [hunk 1]
```diff
@@ -199,5 +195,14 @@
       "prettier --write",
       "git add"
     ]
+  },
+  "jest": {
+    "rootDir": "src",
+    "modulePaths": [
+      "<rootDir>/"
```
Comment (relativityboy) [inline package.json]: not `<rootDir>/src`? 

`https://github.com/steemit/condenser/pull/2235/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R200`

File: package.json [hunk 2]
```diff
@@ -199,5 +195,14 @@
...
+    ],
+    "moduleNameMapper": {
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js"
+    }
   }
 }
```
Comment (relativityboy) [inline package.json]: None of the versions are `^`. In #2235 they were.

File: package.json (hunk 1 above)
Comment (bnchdrff) [reply package.json]: yup; changed the rootDir above in e60d6bf

File: package.json (hunk 2 above)
Comment (bnchdrff) [reply package.json]: correct; we're moving to pinned dependency versions and this is one step in that direction

**Pull request merged**